controllers/front/AddressController.php

```diff
@@ -59,17 +59,25 @@ public function init()
     public function postProcess()
     {
         $this->context->smarty->assign('editing', false);
-        $id_address = (int) Tools::getValue('id_address');
-        // Initialize address if an id exists
+
+        $params = Tools::getAllValues();
+        $id_address = isset($params['id_address']) ? (int) $params['id_address'] : 0;
+
+        // Initialize address if an id exists and prevent fill form with default country second time.
         if ($id_address) {
             $this->address_form->loadAddressById($id_address);
+            // If this is only form request. Adding param id_country for CustomerAddressForm class.
```

Comment:
This looks like a hack honestly, I'll have to look deeper since I'm not sure if we should have to do that there

Reply:
@kpodemski no... this param needs only for correct form fill.